lib/IDE/IDERequests.cpp

```diff
@@ -348,8 +348,8 @@ bool CursorInfoResolver::walkToExprPost(Expr *E) {
     return false;
 
   if (OutermostCursorExpr && isCursorOn(E, LocToResolve)) {
-    CursorInfo = new ResolvedExprStartCursorInfo(
-        CursorInfo->getSourceFile(), CursorInfo->getLoc(), OutermostCursorExpr);
+    CursorInfo = new ResolvedExprStartCursorInfo(CursorInfo->getSourceFile(),
+                                                 CursorInfo->getLoc(), E);
```

Comment:
This change breaks `test/refactoring/RefactoringKind/basic.swift`:

When you have an interpolated string literal and perform cursor info on it, previously we got an interpolated string literal expression. With this change, we get the string literal expression of the first segment inside the string literal before the first interpolation and thus offer to localize the string, which we shouldn’t. I checked and it looks like this change is no longer necessary to get cursor info for literals. Could you revert it?

Reply:
Yes, you are right 👍 I added a test case into the CursorInfo suite as well.

During which I found that getting info of the expression segment in the interpolated string does not work. We consider the whole interpolated string as a single token which has nested tokens within the expression segments. For example this causes that `getSourceRange()` for `ValueDecl` returns such range:

`<START>var foo = <END>"Hello \(42) world"`

This does not work if you are looking for the node of the `42`. Converting to a char based range fixed this.

Reply:
Oh, the good old char source range conversion for interpolated string literals. I so wish this wouldn’t be necessary. 🙄  Good catch though.